packages/website/ts/pages/governance/vote_form.tsx

```diff
@@ -0,0 +1,327 @@
+                </Paragraph>
+                <InputRow>
+                    <Input
+                        name="comments"
+                        label="Leave a private message for the 0x team (Optional)"
+                        type="textarea"
+                        ref={this.commentsRef}
+                        errors={errors}
+                    />
+                </InputRow>
+                {!_.isUndefined(errors.signError) && (
+                    <Paragraph isMuted={true} color={colors.red}>
+                        {errors.signError}
+                    </Paragraph>
+                )}
+                <ButtonRow>
+                    <Button
+                        color="#5C5C5C"
+                        isNoBorder={true}
+                        isTransparent={true}
+                        type="button"
+                        onClick={this.props.onDismiss}
+                    >
+                        Back
+                    </Button>
+                    <ButtonDisabled disabled={!votePreference}>Submit</ButtonDisabled>
+                </ButtonRow>
+            </Form>
+        );
+    }
+    private readonly _createVoteAsync = async (e: FormEvent): Promise<any> => {
+        e.preventDefault();
+
+        const { zeip, votePreference } = this.state;
+        const { selectedAddress, isLedger, providerEngine } = this.props;
+        // Query the available addresses
+        // const addresses = await web3Wrapper.getAvailableAddressesAsync();
+        // Use the first account as the maker
+        // const makerAddress = addresses[0];
+        const makerAddress = selectedAddress;
+        const domainType = [{ name: 'name', type: 'string' }];
+        const voteType = [
+            { name: 'preference', type: 'string' },
+            { name: 'zeip', type: 'uint256' },
+            { name: 'from', type: 'address' },
+            // { name: 'title', type: 'string' },
+        ];
+
+        const domainData = {
+            name: '0x Protocol Governance',
+        };
+        const message = {
+            zeip,
+            preference: votePreference,
+            from: makerAddress,
+            // title: 'MultiAssetProxy: Allow multiple assets per side of a single order',
+        };
+
+        const typedData = {
+            types: {
+                EIP712Domain: domainType,
+                Vote: voteType,
+            },
+            domain: domainData,
+            message,
+            primaryType: 'Vote',
+        };
+
+        const voteHashBuffer = signTypedDataUtils.generateTypedDataHash(typedData);
+        const voteHashHex = `0x${voteHashBuffer.toString('hex')}`;
+        try {
+            // const provider = web3Wrapper.getProvider();
+            const signature = isLedger
+                ? await signatureUtils.ecSignHashAsync(providerEngine, voteHashHex, makerAddress)
+                : await this._eip712SignatureAsync(makerAddress, typedData);
```

Comment:
This will need to fall back to ecSignHashAsync when the provider is `!isLedger` as EIP712 may not be supported in all wallets (Coinbase wallet)